packages/console/src/components/Button/index.tsx

```diff
@@ -32,22 +35,56 @@ const Button = ({
   title,
   icon,
   className,
+  isLoading = false,
+  loadingDelay = 500,
+  onClick,
   ...rest
 }: Props) => {
   const { t } = useTranslation();
+  const [showSpinner, setShowSpinner] = useState(false);
+  const timeout = useRef<number>();
+
+  useEffect(() => {
+    // Delay showing the spinner after 'loadingDelay' milliseconds
+    if (isLoading) {
+      // eslint-disable-next-line @silverhand/fp/no-mutation
+      timeout.current = setTimeout(() => {
+        setShowSpinner(true);
+      }, loadingDelay);
+    } else {
+      clearTimeout(timeout.current);
```

Comment:
seems the `clearTimeout` in return wil be executed already?

Reply:
Yeah, indeed. Updated